pkg-manager/link-bins/src/index.ts

```diff
@@ -251,7 +251,8 @@ async function linkBin (cmd: CommandInfo, binsDir: string, opts?: LinkBinOptions
   if (opts?.preferSymlinkedExecutables && !IS_WINDOWS && cmd.nodeExecPath == null) {
     try {
       await symlinkDir(cmd.path, externalBinPath)
-      await fixBin(cmd.path, 0o755)
+      if (await canFixBin(cmd.path)) await fixBin(cmd.path, 0o755)
+      else globalWarn(`Skipped fixing bin permissions of \`${cmd.path}\` because the file is not owned by the current user`)
     } catch (err: any) { // eslint-disable-line
```

Comment:
What about an optimistic approach? If it fails, then check canFixBin. Also, can't we just check the error?

Reply:
Aaaaa I see. That's true. 
For that though, I'll need to change/remove my test: "linkBins() should not try to change permissions of files not owned by current user"

I'll look into it. Thanks